File: networktables/java/src/main/java/edu/wpi/first/wpilibj/networktables2/TransactionDirtier.java
package edu.wpi.first.wpilibj.networktables2;


/**
 * A transaction receiver that marks all Table entries as dirty in the entry store. Entries will not be passed to the continuing receiver if they are already dirty
 * 
 * @author Mitchell
 *
 */
public class TransactionDirtier implements OutgoingEntryReceiver {
	private final OutgoingEntryReceiver continuingReceiver;
	
	public TransactionDirtier(final OutgoingEntryReceiver continuingReceiver) {
		this.continuingReceiver = continuingReceiver;
	}



	public void offerOutgoingAssignment(NetworkTableEntry entry) {
		if(entry.isDirty())
			return;
		entry.makeDirty();
		continuingReceiver.offerOutgoingAssignment(entry);
	}




	public void offerOutgoingUpdate(NetworkTableEntry entry) {
		if(entry.isDirty())
			return;
		entry.makeDirty();
		continuingReceiver.offerOutgoingUpdate(entry);
	}

}

File: networktables/cpp/parent/src/main/native/networktables2/TransactionDirtier.cpp
/*
 * TransactionDirtier.cpp
 *
 *  Created on: Sep 24, 2012
 *      Author: Mitchell Wills
 */

#include "networktables2/TransactionDirtier.h"



	TransactionDirtier::TransactionDirtier(OutgoingEntryReceiver& _continuingReceiver) : continuingReceiver(_continuingReceiver){
	}



	void TransactionDirtier::offerOutgoingAssignment(NetworkTableEntry* entry) {
		if(entry->IsDirty())
			return;
		entry->MakeDirty();
		continuingReceiver.offerOutgoingAssignment(entry);
	}




	void TransactionDirtier::offerOutgoingUpdate(NetworkTableEntry* entry) {
		if(entry->IsDirty())
			return;
		entry->MakeDirty();
		continuingReceiver.offerOutgoingUpdate(entry);
	}

